pkg/workceptor/kubernetes.go

```diff
@@ -224,6 +237,102 @@ func podRunningAndReady(kw KubeUnit) func(event watch.Event) (bool, error) {
 	return inner
 }
 
+// podInfrastructureSuccess checks if the pod has either successfully started, is pending or is running, or has is successfully terminated.
+// Any other state is considered an infrastructure failure.
+func (ku KubeAPIWrapper) podInfrastructureSuccess(pod *corev1.Pod, containerName string) (bool, string, error) {
+	switch pod.Status.Phase {
+	case corev1.PodFailed:
+		for _, cs := range pod.Status.ContainerStatuses {
+			if cs.State.Terminated != nil && cs.Name == containerName {
```

Comment:
If the containerName is not found we `return true, "", nil` 

I dont think this is what you meant